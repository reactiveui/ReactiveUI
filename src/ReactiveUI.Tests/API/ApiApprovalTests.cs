// Copyright (c) 2020 .NET Foundation and Contributors. All rights reserved.
// Licensed to the .NET Foundation under one or more agreements.
// The .NET Foundation licenses this file to you under the MIT license.
// See the LICENSE file in the project root for full license information.

using System;
using System.Diagnostics.CodeAnalysis;
using System.Linq;
using PublicApiGenerator;
using Shouldly;
using Xunit;

namespace ReactiveUI.Tests.API
{
    [ExcludeFromCodeCoverage]
    public class ApiApprovalTests : ApiApprovalBase
    {
        [Fact]
        public void Testing()
        {
            CheckApproval(typeof(Testing.SchedulerExtensions).Assembly);
        }

        [Fact]
        public void ReactiveUI()
        {
            CheckApproval(typeof(RxApp).Assembly);
        }
    }
}
