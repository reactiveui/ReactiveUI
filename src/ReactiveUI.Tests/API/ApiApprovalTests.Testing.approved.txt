[assembly: System.Runtime.Versioning.TargetFrameworkAttribute(".NETFramework,Version=v4.6.1", FrameworkDisplayName=".NET Framework 4.6.1")]
namespace ReactiveUI.Testing
{
    public class static TestUtils
    {
        public static void AdvanceByMs(this Microsoft.Reactive.Testing.TestScheduler sched, double milliseconds) { }
        public static void AdvanceToMs(this Microsoft.Reactive.Testing.TestScheduler sched, double milliseconds) { }
        public static long FromTimeSpan(this Microsoft.Reactive.Testing.TestScheduler sched, System.TimeSpan span) { }
        public static Microsoft.Reactive.Testing.Recorded<System.Reactive.Notification<T>> OnCompletedAt<T>(this Microsoft.Reactive.Testing.TestScheduler sched, double milliseconds) { }
        public static Microsoft.Reactive.Testing.Recorded<System.Reactive.Notification<T>> OnErrorAt<T>(this Microsoft.Reactive.Testing.TestScheduler sched, double milliseconds, System.Exception ex) { }
        public static Microsoft.Reactive.Testing.Recorded<System.Reactive.Notification<T>> OnNextAt<T>(this Microsoft.Reactive.Testing.TestScheduler sched, double milliseconds, T value) { }
        public static TRet With<T, TRet>(this T sched, System.Func<T, TRet> block)
            where T : System.Reactive.Concurrency.IScheduler { }
        public static void With<T>(this T sched, System.Action<T> block)
            where T : System.Reactive.Concurrency.IScheduler { }
        public static TRet With<TRet>(this ReactiveUI.IMessageBus messageBus, System.Func<TRet> block) { }
        public static void With(this ReactiveUI.IMessageBus messageBus, System.Action block) { }
        public static System.Threading.Tasks.Task<TRet> WithAsync<T, TRet>(this T sched, System.Func<T, System.Threading.Tasks.Task<TRet>> block)
            where T : System.Reactive.Concurrency.IScheduler { }
        public static System.Threading.Tasks.Task WithAsync<T>(this T sched, System.Func<T, System.Threading.Tasks.Task> block)
            where T : System.Reactive.Concurrency.IScheduler { }
        public static System.IDisposable WithMessageBus(this ReactiveUI.IMessageBus messageBus) { }
        public static System.IDisposable WithScheduler(System.Reactive.Concurrency.IScheduler sched) { }
    }
}